### ISSUE TITLE
`--jsx-import-source` option does not work well

### ISSUE THREAD

**Issue by yusukebe** [oven-sh/bun#2905]
### What version of Bun is running?

0.6.1

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

```
bun test --jsx-import-source ../../src/middleware/jsx/jsx-dev-runtime runtime_tests/bun/index.test.tsx
```

### What is the expected behavior?

Does not throw the error.

### What do you see instead?

<img width="920" alt="SS" src="https://github.com/oven-sh/bun/assets/10682/c33e3870-ea95-42d6-8a97-020f1da9720c">

### Additional information

`../../src/middleware/jsx/jsx-dev-runtime` is our custom JSX import source. With v0.5.x, we uses this option, and it works well.

**Comment by dylan-conway** [comment]
this is fixed in #3057 . You will need to change `../../src/middleware/jsx/jsx-dev-runtime` to `../../src/middleware/jsx`. `jsx-dev-runtime` will be resolved from that directory

**Comment by yusukebe** [comment]
Wow, thank you!
